src/Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -296,7 +296,7 @@ private long GetTotalSpeedOfAllPeersBytesPerSec()
         {
             lock (this.peerLock)
             {
-                return this.pullerBehaviorsByPeerId.Sum(x => x.Value.SpeedBytesPerSecond);
+                return this.pullerBehaviorsByPeerId.Where(p => p.Value.SpeedBytesPerSecond > 0).Sum(x => x.Value.SpeedBytesPerSecond);
```

Comment:
Is this really faster?

Reply:
more like ... is this really related to this PR?

Reply:
This fix I can revert.